x-pack/plugins/observability_solution/observability_ai_assistant_app/public/hooks/use_previous_prompts.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { uniq } from 'lodash';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+
+const AI_ASSISTANT_LAST_USED_PROMPT_STORAGE = 'ai-assistant-last-used-prompts';
+
+export function usePreviousPrompts() {
+  const [previousPrompts = [], setPreviousPrompts] = useLocalStorage(
+    AI_ASSISTANT_LAST_USED_PROMPT_STORAGE,
+    [] as string[]
+  );
+
+  return {
+    previousPrompts,
+    addPrompt: (prompt: string) => {
+      if (previousPrompts[0] !== prompt) {
+        // Keep track of the last 5 prompts
+        const newPrompts = uniq([prompt].concat(previousPrompts.filter((_, index) => index < 4)));
+        setPreviousPrompts(newPrompts);
+      }
+    },
```

Comment:
I think this can be simplified as:

`addPrompt: ( prompt:string ) => setPreviousPrompts(( prev ) => uniq([ prompt, ...prev ]).slice(0,5));`

Reply:
@dgieselaar You suggested that earlier, and I responded: https://github.com/elastic/kibana/pull/179696#discussion_r1553738506

With this code:

```
    addPrompt: (prompt: string) =>
      setPreviousPrompts((prev = []) => uniq([prompt, ...prev]).slice(0, 5)),
  };
```

and entering successively 1,2,3,4,5 in a chat, this is the result:

<img width="764" alt="Screenshot 2024-04-08 at 14 04 23" src="https://github.com/elastic/kibana/assets/535564/1939a6c0-65ae-48d5-ae24-f9197ae71b9a">

Reply:
Ah sorry, I didn't see the comment anymore in the file changes. I'm not sure why that bug would happen? How is this different from:
`_.uniq([ '5', ...[ '4', '3', '2', '1', '0' ] ]).slice(0,5)`
which yields: `['5', '4', '3', '2', '1']`

Reply:
Maybe this is because of the `useLocalStorage` hook? I was expecting the same signature as `setState`.